srv/modules/runners/validate.py

```diff
@@ -768,65 +772,67 @@ def ceph_version(self):
-                        continue
-
-            colon_idx = version.find(':')
-            if colon_idx != -1:
-                version = version[colon_idx+1:]
-            dash_idx = version.rfind('-')
-            if dash_idx != -1:
-                version = version[:dash_idx]
-            log.debug("VALIDATE ceph_version: minion ->{}<- final munged version ->{}<-"
-                      .format(minion, version))
-            assert isinstance(version, str), "version value is not a string"
-
-            # "11.10" < "11.2" in Python terms, but not in terms of
-            # version numbering semantics, so we have to break the version number
-            # down into its integer components and compare those separately.
-            #
-            # We can assume that Ceph version numbers will always begin with X.Y.Z
-            # where X, Y, and Z are integers. Here, we are only interested in X and Y.
-            #
-            # In other words, there must be at least two version components and
-            # both must be convertible into integers.
-
-            if not all(s.isdigit() for s in version.split(".")[0:2]):
-                failmsg = ("Minion {} reports unparseable Ceph version {}"
-                           .format(minion, version))
-                if self.in_dev_env:
-                    log.warning('VALIDATE ceph_version: ' + failmsg)
-                else:
-                    self.errors.setdefault('ceph_version', []).append(failmsg)
-                    continue
+                    # Something is really wrong
+                    prefix = 'Version missing from'
+                    self.errors.setdefault('ceph_version', [prefix]).append(minion)
+            else:
+                self.uninstalled.append(minion)
 
-            if LooseVersion(version) < LooseVersion(LUMINOUS_VERSION):
-                self.errors.setdefault('ceph_version', []).append(
-                    "The Ceph version available on minion {} ({}) is older than 'luminous' ({})"
-                    .format(minion, version, LUMINOUS_VERSION))
+    def _check_available(self):
+        """
+        Check for available Ceph packages.  If all minions have Ceph installed,
+        then the query has no results.
+        """
+        if not self.uninstalled:
+            return
 
-        self._set_pass_status('ceph_version')
+        search = "L@{}".format(",".join(self.uninstalled))
+        # When search matches no minions, salt prints to stdout.  Suppress stdout.
+        _stdout = sys.stdout
+        sys.stdout = open(os.devnull, 'w')
+        results = self.local.cmd(search, 'pkg.info_available', [self.package],
+                                 tgt_type="compound")
+        sys.stdout = _stdout
+        for minion in results:
+            if isinstance(results[minion], dict) and self.package in results[minion]:
+                if 'version' in results[minion][self.package]:
+                    version = results[minion][self.package]['version']
```

Comment:
and here, could we have `log.debug("VALIDATE ceph_version: minion ->{}<- info_available version ->{}<-".format(minion, version))` please